src/bgw/job_stat.c

```diff
@@ -562,7 +598,34 @@ ts_bgw_job_stat_next_start(BgwJobStat *jobstat, BgwJob *job, int32 consecutive_f
 		return DT_NOBEGIN;
 
 	if (jobstat->fd.consecutive_crashes > 0)
+	{
+		/* Update the errors table regarding the crash */
+		if (!ts_flags_are_set_32(jobstat->fd.flags, LAST_CRASH_REPORTED))
+		{
+			/* add the proc_schema, proc_name to the jsonb */
+			NameData proc_schema = { 0 }, proc_name = { 0 };
+			namestrcpy(&proc_schema, NameStr(job->fd.proc_schema));
+			namestrcpy(&proc_name, NameStr(job->fd.proc_name));
+			JsonbParseState *parse_state = NULL;
+			pushJsonbValue(&parse_state, WJB_BEGIN_OBJECT, NULL);
+			ts_jsonb_add_str(parse_state, "proc_schema", NameStr(proc_schema));
+			ts_jsonb_add_str(parse_state, "proc_name", NameStr(proc_name));
+			JsonbValue *result = pushJsonbValue(&parse_state, WJB_END_OBJECT, NULL);
+
+			const FormData_job_error jerr = {
+				.error_data = JsonbValueToJsonb(result),
+				.start_time = jobstat->fd.last_start,
+				.finish_time = ts_timer_get_current_timestamp(),
+				.pid = -1,
```

Comment:
Is there a way to get the pid here?

Reply:
I think the most straightforward way is to update the `bgw_job_stat` table once the job starts, for example in a field `pid integer`. Maybe it should be `pid bigint` so I can use 32 bits for the pid and the rest for the flags. Then again, I think pids will be way less than 2^32 so it could be safe to use the `flags` field to store both the pid and the flag `last_crash_reported`